pygmt/helpers/testing.py

```diff
@@ -139,3 +140,36 @@ def wrapper(*args, ext="png", request=None, **kwargs):
         return wrapper
 
     return decorator
+
+
+def download_test_data():
+    """
+    Convenience function to download remote data files used in PyGMT tests.
+    """
+    # List of datasets to download
+    datasets = [
+        # Earth relief grids
+        "@earth_relief_01d_p",
+        "@earth_relief_01d_g",
+        "@earth_relief_30m_p",
+        "@earth_relief_30m_g",
+        "@earth_relief_10m_p",
+        "@earth_relief_10m_g",
+        "@earth_relief_05m_p",
+        "@earth_relief_05m_g",
+        # List of tiles of 03s/01s srtm data.
+        # Names like @N35E135.earth_relief_03s_g.nc is for internal use only.
+        # The naming scheme may change. DO NOT USE IT IN YOUR SCRIPTS.
+        "@N35E135.earth_relief_03s_g.nc",
+        # Other cache files
+        "@fractures_06.txt",
+        "@ridge.txt",
+        "@srtm_tiles.nc",  # needed for 03s and 01s relief data
+        "@Table_5_11.txt",
+        "@test.dat.nc",
+        "@tut_bathy.nc",
+        "@tut_quakes.ngdc",
+        "@tut_ship.xyz",
+        "@usgs_quakes_22.txt",
+    ]
+    which(fname=datasets, download="a")
```

Comment:
Should we also return a list to the downloaded datasets?

Reply:
Sounds good. BTW, should I temporarily trigger the `cache_data.yml` workflow to make sure that it works as expected?

Reply:
https://github.com/GenericMappingTools/pygmt/pull/1310/checks?check_run_id=2724433035

The cache_data.yml workflow fails, because PyGMT is not installed in this CI job.

Reply:
https://github.com/GenericMappingTools/pygmt/pull/1310/checks?check_run_id=2724622181

Now the cache_data.yml workflow works as expected.